src/scale.ts

```diff
@@ -22,14 +22,17 @@ export namespace ScaleType {
   export const ORDINAL_LOOKUP: 'ordinal' = 'ordinal';
   export const POINT: 'point' = 'point';
   export const BAND: 'band' = 'band';
+
+  // Ordinal color scale
+  export const INDEX: 'index' = 'index';
```

Comment:
Need to update `scaleTypeSupportProperty` at the end of this file and write test.